nntrainer/layers/layer_node.cpp

```diff
@@ -474,6 +411,25 @@ InitLayerContext LayerNode::finalize(const std::vector<TensorDim> &input_dims) {
   if (run_context)
     throw std::runtime_error("Finalizing a layer which is already finalized");
 
+  /** manipulate layers if required */
+  if (getType() == ActivationLayer::type) {
+    auto &act_prop = std::get<props::Activation>(*layer_node_props_realization);
+    if (!act_prop.empty()) {
+      layer->setProperty({"activation=" + to_string(act_prop)});
+    }
+  }
+  if (getType() != TimeDistLayer::type && getDistribute()) {
+    std::unique_ptr<TimeDistLayer> dlayer(new TimeDistLayer());
```

Comment:
If we do this here, is it ok? I mean the order of execution is already set before this layer is finalized.

Reply:
dist layer changes at least one property of the layer - supportInPlace(). If this is done here, supportInPlace must be returned simply not by the layer, but by the dist layer even before finalize (as inPlaceOptimization has been moved to compile time).

Reply:
Time dist layer does not change the order of execution so it should be okay.

> dist layer changes at least one property of the layer - supportInPlace(). If this is done here, supportInPlace must be returned simply not by the layer, but by the dist layer even before finalize (as inPlaceOptimization has been moved to compile time).

There is two (potential) bugs related with this.

1. `getLayer()` is returning inside layer, supportInPlace is not queried.
```
nntrainer::Layer *LayerNode::getLayer() {
  if (getDistribute())
    return ((TimeDistLayer *)(layer.get()))->getDistLayer();
  else
    return layer.get();
}
```

2. We will need another end point just like finalize() to invalidate changing properties after connection has been set. 

This will need more fundamental changes but I'll make a quick fix out of this PR.